File: backend/domain/consultant.py
from dataclasses import dataclass
from datetime import datetime

from enums import Batch, PlacementStatus


@dataclass(slots=True)
class Consultant:
    user_id: int
    email: str
    first_name: str
    last_name: str
    created_at: datetime
    id: int | None = None
    batch: Batch | None = None
    placement_status: PlacementStatus = PlacementStatus.ONBOARDING
    client: str | None = None

File: frontend/src/types/consultant.ts
export type Batch = "PYTHON" | "JAVA" | "DATA" | "ANDROID";

export type PlacementStatus =
  | "ONBOARDING"
  | "TRAINING"
  | "AVAILABLE"
  | "PLACED"
  | "ENDING_SOON";

export interface UnassignedUser {
  id: number;
  email: string;
  first_name: string;
  last_name: string;
}

export interface CreateConsultantRequest {
  user_id: number;
  batch: Batch;
  placement_status: PlacementStatus;
  client: string | null;
}

export interface Consultant {
  id: number;
  user_id: number;
  email: string;
  first_name: string;
  last_name: string;
  batch: Batch;
  placement_status: PlacementStatus;
  client: string | null;
}